### ISSUE TITLE
Child nodes not set to empty list/null

### ISSUE THREAD

**Issue by GoogleCodeExporter** [Kobee1203/jcrom#87]
```
Hi

For us an issue has come up, which might not affect most users, but I 
thought I post it anyway:

We have added an object caching mechanism to JCROM, and I just had the 
problem that when refreshing an existing object, the last child object 
would not get removed from the list, although the child-node in JCR was 
deleted.

This is caused by the ChildNodeMapper class, which only sets the child 
instances field if there are any existing nodes (plus a few exceptions). 
Basically I have changed the last lines of that file to:

        }
        else {
            // Tom: set field to null, to remove any previously 
existing children!
            field.set(obj, null);
        }
    }
}

```

Original issue reported on code.google.com by `Thomas.A...@gmail.com` on 21 Dec 2009 at 2:47
